trade/src/main/java/bisq/trade/Trade.java

```diff
@@ -92,6 +94,10 @@ private static TradeRole createRole(boolean isBuyer, boolean isTaker) {
     // transient fields are excluded by default for EqualsAndHashCode
     private transient final TradeRole tradeRole;
     private final C contract;
+    @Setter
+    private ErrorCode errorCode = ErrorCode.UNSPECIFIED;
```

Comment:
If we add it here we should also add it to the protobuf to have it persisted. But I will look into alternative implementations.